### DISCUSSION TITLE
Skill ranges for individual role sets + Separation of fabrication skills from practical skills

### DISCUSSION THREAD

**Discussion by Solbusaur** [FakeFishGames/Barotrauma#4786] [Ideas and suggestions]
We could benefit from role sets having individual skill ranges, examples being the gunner security officer. The fact that they spawn with a wrench should suggest they are more adept at mechanical repairs than your average law enforcer, yet their mechanical skills are limited to the same as every other security set. There could be a lot fewer cut hands for gunners, and a shorter levelling up time, if this change was made.

Another example, and leading into my second point, is the first responder and medical doctor variants. Your frontline medic would be more fitted for fast and efficient healing of a patient whilst the regular medical doctor, whilst still trained for recovery, should be a lot more proficient in the development of medical items. I would consider separating weapons, electrical, mechanical and medical crafting skills into a single fabricator skillset.
For an item to be crafted, both the fabrication skill and the [role] skill thresholds must be met if it is to be crafted efficiently.
Now the frontline medic has to work harder with a fabricator if they are to be as efficient as their stay-at-home counterpart, and a primarily crafting-oriented mechanic isn't as efficient as repairs as the boiler room junkie.
Perhaps the gunner security officer could also have a raised fabricator skill for the easier creation of more ammunition.

**Comment by RadCatish** [comment]
I agree! It's a little bizarre that our mechies always end up with really high weapons skills from fabricating ammo, despite rarely manning the guns

**Reply by Kulik242** [reply]
It makes sense alright, but gameplay wise I'm against deeper specialization. It would mean that you need more players to cover all the roles on the ship or make trade offs that hamper the crew's effectiveness.

**Comment by NilanthAnimosus** [comment]
I have a few thoughts on this,

Given my average life expectancy in game I sort of prefer skills to be merged and have multiple uses as it currently is now, At most - a general proficiency for fabrication could be added and have various machines be effected by 2 separate skills, so fabrication in general makes you better at other skills for fabrication speed partially, but Electrical becoming "Only electrical repairs" and medical "Only healing" and crafting a ton of metal storage containers making you a genius at the most complicated medical substances are two very different things.

I do think different roles in the same job should be allowed different skill ranges and show these on the load out preview, this is more for modding purposes as I cannot think of too many reasons to change the vanilla load outs on these, 

If one wanted to make changes to skills, I would keep the existing skills as-is and at most add general fabrication as a skill (But keep requirements for crafting items based solely off their core skill, fabrication only aiding the time from that requirement),  and have skills able to have an innate bonus to the rate they gain skill (A multiplier so a medic gains medical faster, mechanics earn mechanical faster, and such - sometimes I end up a better engineer as a medic than I am actually at medical skills which is just funny to me), you think there would be a skillgain malus for skills far outside your chosen job, and a bonus for those inside it, with some being neutral (such as helm unless captain).
